code/go/0chain.net/smartcontract/storagesc/writepool.go

```diff
@@ -299,9 +301,12 @@ func (ssc *StorageSmartContract) getWritePool(clientID datastore.Key,
 	return
 }
 
-func (ssc *StorageSmartContract) createWritePool(t *transaction.Transaction,
-	alloc *StorageAllocation, balances chainState.StateContextI) (err error) {
-
+func (ssc *StorageSmartContract) createWritePool(
+	t *transaction.Transaction,
+	alloc *StorageAllocation,
+	force bool,
```

Comment:
not nice. maybe just don't transfer that "undercapitalized" locks and leave them as is.
as a result there will be no need to patch this and `fill` funcs.

Reply:
Yes, still undecided about how to write this; not sure it's possible to not write it in a not nice fashion. This and some unit tests to add is mainly why it is WIP.